service/slice_config_webhook_validation_test.go

```diff
@@ -1420,7 +1426,135 @@ func ValidateMaxClusterForParticipatingCluster(t *testing.T) {
 	require.Contains(t, err.Error(), "participating clusters cannot be greater than MaxClusterCount")
 	clientMock.AssertExpectations(t)
 }
+func TestValidateCertsRotationInterval_Postive(t *testing.T) {
+	name := "slice_config"
+	namespace := "randomNamespace"
+	clientMock, sliceConfig, ctx := setupSliceConfigWebhookValidationTest(name, namespace)
+	sliceConfig.Spec.RenewBefore = metav1.Now()
+	expiry := metav1.Now().Add(30)
+	clientMock.On("Get", ctx, mock.Anything, mock.Anything).Return(nil).Run(func(args mock.Arguments) {
+		arg := args.Get(2).(*controllerv1alpha1.VpnKeyRotation)
+		arg.ObjectMeta = metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		}
+		arg.Spec = controllerv1alpha1.VpnKeyRotationSpec{
+			SliceName:               name,
+			CertificateCreationTime: metav1.Now(),
+			CertificateExpiryTime:   metav1.Time{Time: expiry},
+		}
+	}).Once()
+	oldSliceConfig := controllerv1alpha1.SliceConfig{}
+	err := validateRenewNowInSliceConfig(ctx, sliceConfig, &oldSliceConfig)
+	require.Nil(t, err)
+}
+
+func TestValidateCertsRotationInterval_Negative(t *testing.T) {
+	name := "slice_config"
+	namespace := "randomNamespace"
+	clientMock, sliceConfig, ctx := setupSliceConfigWebhookValidationTest(name, namespace)
+	// RenewBefore is 1 hour after, decline
+	sliceConfig.Spec.RenewBefore = metav1.Time{Time: metav1.Now().Add(time.Hour * 1)}
+	expiry := metav1.Now().Add(30)
+	clientMock.On("Get", ctx, mock.Anything, mock.Anything).Return(nil).Run(func(args mock.Arguments) {
+		arg := args.Get(2).(*controllerv1alpha1.VpnKeyRotation)
+		arg.ObjectMeta = metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		}
+		arg.Spec = controllerv1alpha1.VpnKeyRotationSpec{
+			SliceName:               name,
+			CertificateCreationTime: metav1.Now(),
+			CertificateExpiryTime:   metav1.Time{Time: expiry},
+		}
+	}).Once()
+	oldSliceConfig := controllerv1alpha1.SliceConfig{}
+	err := validateRenewNowInSliceConfig(ctx, sliceConfig, &oldSliceConfig)
+	require.NotNil(t, err)
+}
+
+func TestValidateCertsRotationInterval_NegativeClusterStatus(t *testing.T) {
+	name := "slice_config"
```

Comment:
This story was not closed as the ITs were not created in the last two sprints but still we have the same issue
We need have the ITs to cover the Key Rotation Interval checks